llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -936,6 +937,51 @@ Error MCCASBuilder::createStringSection(
   return Error::success();
 }
 
+/// Reads and returns the length field of a DWARF header contained in
+/// Reader, assuming Reader is positioned at the beginning of the section.
+/// The Reader's state is advanced to the first byte after the section.
+static Expected<uint64_t>
+getSizeFromDWARFHeaderAndSkip(BinaryStreamReader &SectionReader) {
+  // The DWARF headers have a standard format, if it is 32-bit DWARF, the first
+  // 4 bytes are a length field which we want to read and then skip over to get
+  // to the end of a function's contribution to the particular DWARF Section. If
+  // the first 4 bytes are 0xffffffff it is 64-bit DWARF and the next 8 bytes
+  // contain the actual length that we then skip.
+  uint32_t Length;
+  if (auto E = SectionReader.readInteger(Length))
+    return E;
+  if (Length != 0xffffffff) {
+    // The Debug Information is 32-bit DWARF
+    if (auto E = SectionReader.skip(Length))
+      return E;
+    return Length;
+  }
+  uint64_t ExtendedLength;
+  if (auto E = SectionReader.readInteger(ExtendedLength))
+    return E;
+  if (auto E = SectionReader.skip(ExtendedLength))
+    return E;
+  return ExtendedLength;
```

Comment:
I understand the desire to account for the 64 bit format, but right now this code is both unreachable and untestable, so IMO we should remove it

Reply:
I don't think it is hurting anything and it is correct, it is not like this will change. So why rewrite this in the future?

Reply:
> it is correct

I will defer to others here, but IMO code that is not tested cannot be said to be correct

Reply:
> but IMO code that is not tested cannot be said to be correct

Yeah, let's either stick an llvm_unreachable there, or find a way to test it.

Reply:
It has been changed with the rebase :)